### PR TITLE
Checkout info

### PR DESCRIPTION
**Pull request by magda-woj**
Demo-app: implemented a buyer's details screen after clicking checkout button in cart.

### CODE REVIEW

**Comment by breedx-splk** [comment]
Nice! So after clicking on "Checkout" in the cart, the user is taken to this screen:

image

File: demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/cart/CheckoutInfo.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+}
+
+fun <T : Any> T.allFieldsNotBlank(): Boolean {
+    return this::class.memberProperties.all { property ->
+        property.isAccessible = true
+        val value = property.call(this)
+        value is String && value.isNotBlank()
+    }
+}
```
Comment (breedx-splk) [inline demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/cart/CheckoutInfo.kt:148]: I think we should avoid this reflection and instead build explicit methods on each of the data classes. Something like this:

```
fun isComplete(): Boolean {
    return arrayOf(email, streetAddress, zipCode, city, state, country)
        .all { it.isNotBlank() }
}
```

The reflection feels a bit forced, and it modifies the permission/accessibility in an unwanted way. It also limits the ability add an optional field (like address line 2 or whatever) down the road.

Comment (marandaneto) [reply demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/cart/CheckoutInfo.kt:148]: also if minified is enabled, the methods might be removed unless its added in the proguard file

**Review by LikeTheSalad — APPROVED** [review]

**Review by breedx-splk — APPROVED** [review]
Solid, thanks!

**Pull request merged**